spec/rubocop/cop/lint/literal_as_condition_spec.rb

```diff
@@ -2,49 +2,119 @@
 
 RSpec.describe RuboCop::Cop::Lint::LiteralAsCondition, :config do
   %w(1 2.0 [1] {} :sym :"#{a}").each do |lit|
-    it "registers an offense for literal #{lit} in if" do
+    it "registers an offense for truthy literal #{lit} in if" do
       expect_offense(<<~RUBY, lit: lit)
         if %{lit}
            ^{lit} Literal `#{lit}` appeared as a condition.
           top
         end
       RUBY
+
+      expect_correction(<<~RUBY)
+        top
+      RUBY
+    end
```

Comment:
I think that `#{a}` should not be included in the correctable cases because unlike the other literals, removing it will potentially stop calling a method which could have side effects.